### PR TITLE
BUGFIX: SD-209 should not have middle zero when min > 0 or 0 > max

### PR DESCRIPTION
**Pull request by naibinh**

### CODE REVIEW

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2710/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2091/ : SUCCESS in 7m 47s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2712/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2092/ : SUCCESS in 2m 59s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1112/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2713/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/657/ : SUCCESS in 19m 07s

File: src/components/visualizations/chart/highcharts/plugins/test/getTickPositions.spec.ts [hunk 1]
```diff
@@ -517,6 +517,40 @@ describe('get tick positions for dual axes chart', () => {
                 });
             });
 
+            it('left tick position is empty with undefined min/max,' +
```
Comment (strouhaljan) [inline src/components/visualizations/chart/highcharts/plugins/test/getTickPositions.spec.ts]: Let's use newer style of describing tests:
1) test description starts with the word `should` in order to improve readability => `it('should...`
2) please keep description on single line and rather add exception for linter `// eslint-disable-next-line`

File: src/components/visualizations/chart/highcharts/plugins/zeroAlignYAxis.ts [hunk 2]
```diff
@@ -299,7 +303,11 @@ export function getTickPositioner() {
                 if (isNil(axis.min) || isNil(axis.max)) {
                     // this axis could be not initiated (undefined) or could be invalid (null)
                     const isOtherAxisInvalid = minmax[yAxisIndex] === null;
-                    return (isOtherAxisInvalid && !isLineChartType) ? // line chart does not need middle zero
+
+                    const isRangeWithMiddleZero = isOtherAxisInvalid &&
```
Comment (strouhaljan) [inline src/components/visualizations/chart/highcharts/plugins/zeroAlignYAxis.ts]: Please use only one level indentation on second a third line of expression:
```
    const isRangeWithMiddleZero = isOtherAxisInvalid &&
        !isLineChartType && // line chart does not need middle zero
        isMinMaxCoverZero(min, max);
```

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2823/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2199/ : SUCCESS in 1m 41s

File: src/components/visualizations/chart/highcharts/plugins/test/getTickPositions.spec.ts [hunk 3]
```diff
@@ -555,8 +589,8 @@ describe('get tick positions for dual axes chart', () => {
                 });
             });
 
-            it('left tick position is empty with out-of-range min/max,' +
-                'right tick position is empty with min > max,', () => {
+            // tslint:disable-next-line:max-line-length
+            it('should left tick position be empty with out-of-range min/max, right tick position be empty with min > max,', () => {
```
Comment (strouhaljan) [inline src/components/visualizations/chart/highcharts/plugins/test/getTickPositions.spec.ts]: This sentence doesn't make sense in english :(
What about something like: `it('should return empty left tick position if min/max is out of range; and empty right tick position if min > max')`?
Please review recommended sentence above if it makes sense in this context.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2858/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2231/ : SUCCESS in 2m 04s

**Comment by naibinh** [comment]
@strouhaljan , I have resolves your comments.
Please check it and thank you.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2861/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2234/ : SUCCESS in 11m 41s

**Comment by naibinh** [comment]
In this new commit, I updated description follow your comments and also updated some descriptions that I think it should be.

So do you mean to revert all my own updates or all (include yours and mine)?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2871/ : SUCCESS in 36s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2246/ : SUCCESS in 36m 12s

**Comment by strouhaljan** [comment]
lgtm

**Comment by naibinh** [comment]
@strouhaljan , I did it.
Thanks.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2879/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2254/ : SUCCESS in 8m 34s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/621/ : SUCCESS in 34s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2257/ : SUCCESS in 6m 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1186/ : SUCCESS in 24m 20s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/593/ : SUCCESS in 22m 06s

**Pull request merged**